lib/facter/resolvers/linux/networking.rb

```diff
@@ -31,11 +31,16 @@ def add_info_from_socket_reader
             @fact_list[:interfaces].keys.each do |interface_name|
               mtu(interface_name, mtu_and_indexes)
               dhcp(interface_name, mtu_and_indexes)
+              operstate(interface_name)
 
               @log.debug("Found interface #{interface_name} with #{@fact_list[:interfaces][interface_name]}")
             end
           end
 
+          def operstate(interface_name)
+            @fact_list[:interfaces][interface_name][:operational_state] = File.read("/sys/class/net/#{interface_name}/operstate").strip
```

Comment:
This could raise if the file doesn't exist, permissions problem, etc. Suggest using the FileHelper:

```suggestion
            state = Facter::Util::FileHelper.safe_read("/sys/class/net/#{interface_name}/operstate", nil)
            @fact_list[:interfaces][interface_name][:operational_state] = state.strip if state
```

Reply:
Does this look better?